main.tf

```diff
@@ -0,0 +1,173 @@
+data "aws_iam_policy_document" "assume_role" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+data "aws_iam_policy_document" "lambda_logging" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+    ]
+
+    resources = ["arn:aws:logs:*:*:*"]
+  }
+}
+
+data "aws_iam_policy_document" "lambda_s3_permissions" {
+  statement {
+    effect = "Allow"
+
+      actions = [
+        "s3:*",
+        "s3-object-lambda:*",
+      ]
+
+      resources = ["arn:aws:s3:::*/*"]
+  }
+}
+
+data "aws_iam_policy_document" "json_bucket_topic" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["s3.amazonaws.com"]
+    }
+
+    actions   = ["SNS:Publish"]
+    resources = ["arn:aws:sns:us-east-1:380255901104:JSON_to_CSV_conversion_complete"]
```

Comment:
Instead of hardcoding, try to refer the arn from the terraform resource block where it is created